src/net_processing.cpp

```diff
@@ -2720,6 +2725,25 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                 peer->m_tx_relay->m_relay_txs = fRelay; // set to true after we get the first filter* message
             }
             if (fRelay) pfrom.m_relays_txs = true;
+
+            if (greatest_common_version >= WTXID_RELAY_VERSION && m_txreconciliation) {
+                // Per BIP-330, we announce reconciliation support if:
+                // - protocol version per the VERSION message supports for WTXID_RELAY
+                // - the peer indicated support for transaction relay in the VERSION message
+                // - we are not in -blocksonly mode
+                if (pfrom.m_relays_txs && !m_ignore_incoming_txs) {
+                    const uint64_t recon_salt = m_txreconciliation->PreRegisterPeer(pfrom.GetId());
+                    // Only the inbound peer initiate reconciliations and the outbound peer is
+                    // supposed to only respond. Here we suggest our roles in reconciliations
+                    // (initiator/responder) based on the connection direction.
+                    // If the way we assign the roles is updated, reconciliation protocol
```

Comment:
In "p2p: Announce reconciliation support"

This seems inconsistent to me. The BIP330 document defines the reconciliation protocol version 1, but I don't think it says anything about which side is supposed to initiate (and the presence of the initiator/responder bools seems to be designed to explicitly support both).

Is there a reason for requiring a recon protocol version update when the role assignment changes?
* If not, just drop the comment.
* If there is, it seems cleaner to have BIP330 just hardcode v1 = outbound-initiated, and forego the two booleans. Later versions (in potential future BIPs) can change that.